backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts

```diff
@@ -59,4 +152,197 @@ describe("POST /credential", () => {
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given the access token signature could not be verified", () => {
+    it("Returns 500 Server Error", async () => {
+      const accessTokenWithInvalidSignature = makeSignatureUnverifiable(
+        accessToken,
+        "6T5a8kCTyXsmw_2ATkyPgtLRzsuot-_ZIXWnuXNftZP8SHHkNxwFyMaZxEnqqtQst-99AoRrUDZnPov0oztbSA",
+      );
+
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessTokenWithInvalidSignature,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "server_error",
+        error_description: "Server Error",
+      });
+      expect(response.status).toBe(500);
+    });
+  });
+
+  describe("Given the request is valid and an active session is found for a given sub", () => {
+    it("Returns 200 OK", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessToken,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        "https://vocab.account.gov.uk/v1/credentialStatus": "pending",
+        sub: "urn:fdc:gov.uk:2022:56P4CMsGh_02YOlWpd8PAOI-2sVlB2nsNU7mcLZYhYw=",
+      });
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("Given the same access token is used more than once to fetch an active session", () => {
+    it("Returns 200 OK", async () => {
+      const response = await axiosInstance.post(
+          `${apiBaseUrl}/async/credential`,
+          credentialRequestBody,
+          {
+            headers: {
+              "X-Custom-Auth": "Bearer " + accessToken,
+            },
+          },
+      );
+
+      expect(response.data).toStrictEqual({
+        "https://vocab.account.gov.uk/v1/credentialStatus": "pending",
+        sub: "urn:fdc:gov.uk:2022:56P4CMsGh_02YOlWpd8PAOI-2sVlB2nsNU7mcLZYhYw=",
+      });
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("Given the request is valid and a new session is created for a new sub", () => {
+    it("Returns 201 Created", async () => {
+      const randomSub = randomUUID();
+      const credentialRequestBody = getRequestBody(clientDetails, randomSub)
+
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+          credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessToken,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        "https://vocab.account.gov.uk/v1/credentialStatus": "pending",
+        sub: randomSub,
+      });
+      expect(response.status).toStrictEqual(201);
+    });
+  });
 });
+
+interface ClientDetails {
```

Comment:
I can move these helpers somewhere else but I'm just a bit concerned about creating too many helpers/utils files...

Reply:
if the helpers are for the file in question - let's keep them co-located. I despise a 'utils' folder - we can have a long chat on this one over a ☕ 😄